crates/voicevox_core_c_api/src/lib.rs

```diff
@@ -670,9 +675,12 @@ pub unsafe extern "C" fn voicevox_wav_free(wav: *mut u8) {
 pub extern "C" fn voicevox_error_result_to_message(
     result_code: VoicevoxResultCode,
 ) -> *const c_char {
-    BUFFER_MANAGER.memorize_static_str(
-        voicevox_core::result_code::error_result_to_message(result_code).as_ptr() as *const c_char,
+    let message = CStr::from_bytes_with_nul(
+        voicevox_core::result_code::error_result_to_message(result_code).as_ref(),
     )
+    .expect("`error_result_to_message` should always return NUL-terminated string");
```

Comment:
ただの`.unwrap()`からメッセージを書いてみました。

Reply:
特に取り決めないですが、日本語だと統一感出るかも？
（suggestionしようとしたけど日本語訳よく知りませんでした。。）

Reply:
「ヌル終端文字列」ですかね。[rust-jp](https://rust-jp.rs/)だと「ヌル終端」にすることにしてたと思うので。

```suggestion
    .expect("`error_result_to_message`はヌル終端文字列を返すはずである");
```

(ちなみに[Pythonだと"zero-terminated strin"/"ゼロ終端文字列"](https://docs.python.org/3/library/ctypes.html#ctypes.c_char_p)らしいです)

今並行してDoxygenとSphinxのドキュメントを全面的に書き直そうと思っているのですが、そっちでの「ヌル終端」で統一しようと思ってます。
(ちなみにこんな感じにしようかと思ってます)

![image](https://github.com/VOICEVOX/voicevox_core/assets/14125495/4d1261b6-5e69-47b8-aec0-de2e9ea29916)

Reply:
良さそうに思いました！　suggestion取り込まさせていただきます 🙏